src/test/java/bio/terra/service/filedata/google/firestore/FireStoreDaoTest.java

```diff
@@ -22,56 +26,73 @@
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
 import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
 import org.springframework.test.context.ActiveProfiles;
-import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
 
-@RunWith(SpringRunner.class)
+@ExtendWith(SpringExtension.class)
 @SpringBootTest
 @AutoConfigureMockMvc
 @ActiveProfiles({"google", "connectedtest"})
-@Ignore
-@Category(Connected.class)
+@Tag(Connected.TAG)
 @EmbeddedDatabaseTest
-public class FireStoreDaoTest {
-  private final Logger logger = LoggerFactory.getLogger(FireStoreDaoTest.class);
-
+class FireStoreDaoTest {
   @Autowired private FireStoreDirectoryDao directoryDao;
-
   @Autowired private FireStoreFileDao fileDao;
-
   @Autowired private FireStoreDao dao;
-
   @Autowired private FireStoreUtils fireStoreUtils;
-
   @Autowired private FireStoreDependencyDao fireStoreDependencyDao;
+  @Autowired private ConnectedOperations connectedOperations;
+  @Autowired private ConnectedTestConfiguration testConfig;
+  @MockBean private IamProviderInterface samService;
+  @Autowired private ConfigurationService configService;
 
   private Firestore firestore;
-  private UUID datasetId;
-  private UUID snapshotId;
+  private String datasetId;
+  private String snapshotId;
+  private Dataset dataset;
+  private Snapshot snapshot;
 
-  @Before
+  @BeforeEach
   public void setup() throws Exception {
-    firestore = TestFirestoreProvider.getFirestore();
-    datasetId = UUID.randomUUID();
-    snapshotId = UUID.randomUUID();
+    connectedOperations.stubOutSamCalls(samService);
+    configService.reset();
+
+    // Create dataset so that we have a firestore instance to test with
+    BillingProfileModel billingProfile =
+        connectedOperations.createProfileForAccount(testConfig.getGoogleBillingAccountId());
+    DatasetSummaryModel summaryModel =
+        connectedOperations.createDataset(billingProfile, "dataset-minimal.json");
+    GoogleProjectResource projectResource =
+        new GoogleProjectResource().googleProjectId(summaryModel.getDataProject());
+    dataset = new Dataset().id(summaryModel.getId()).projectResource(projectResource);
+    datasetId = summaryModel.getId().toString();
+    var snapshotIdUUID = UUID.randomUUID();
+    snapshotId = snapshotIdUUID.toString();
+    snapshot = new Snapshot().id(snapshotIdUUID).projectResource(projectResource);
+
+    // real case will have separate dataset and snapshot instances
+    // But, we can share this firestore instance for this test
+    firestore = TestFirestoreProvider.getFirestore(summaryModel.getDataProject());
```

Comment:
Have you considered rewriting these tests as unit tests following our [unit testing approach](https://broadworkbench.atlassian.net/wiki/spaces/IRT/pages/2498494502/All+Roads+Lead+to+Unit+Tests)? This is a suggestion that would require a much larger change, but I figure I would suggest it since you are already doing work in the connected tests.

Concretely, I am suggesting that we would isolate the few critical `Firestore` class operations that we execute into a class and/or test (e.g. `firestore.runTransaction`, `firestore.collection`, `firestore.document`, ...) and these would be our new "connected" or "red ring" tests. Layers above that rely on these critical operations would use a mocked version of these abstractions instead.

This would enable us to increase our overall test coverage and confidence by relying more on unit tests and decreasing our reliance on connected tests. As the unit testing approach discusses, connected (or integration) tests require us to write a multiplicative number of tests for each code path, instead of a linear number with unit tests. We spend about ~35m running Firestore connected tests in total, which is a decent chunk of time.

Overall I think the effort would be high, but the value would be high, too. Also, we don't have to do everything at once; an intermediate compromise could be to start unit testing some of the most critical components to help us continue to build test confidence.

Reply:
I'm in favor of this approach -- but, I think it would be fair to treat that migration as a separate body of work rather than blocking this PR, evaluated and prioritized relative to any competing priorities.

> Also, we don't have to do everything at once; an intermediate compromise could be to start unit testing some of the most critical components to help us continue to build test confidence.

☝️ I like this incremental value mindset though!

Reply:
Yeah, thanks for bringing this up. I also thought about trying to migrate these to unit tests. However, these tests test the very basic firestore operations. I think some of the higher level firestore connected tests would be better suited to mocking. But, it would be good to take a broader look the firestore connected tests, given the high percentage of the test time.
I would like to move ahead with the changes in this PR, but definitely am open to seeing how we can move towards more unit tests with future work.